packages/lodestar/src/network/gossip/validator.ts

```diff
@@ -113,104 +135,119 @@ export class GossipMessageValidator implements IGossipMessageValidator {
+      this.logger.info("Received valid gossip aggregate and proof", logContext);
+      return ExtendedValidatorResult.accept;
     } catch (e) {
       this.logger.error("Error while validating gossip aggregate and proof");
+      if (!(e instanceof AttestationError)) {
+        this.logger.error("Gossip aggregate and proof validation threw a non-AttestationError", e);
+        return ExtendedValidatorResult.ignore;
+      }
       switch (e.type.code) {
         case AttestationErrorCode.ERR_WRONG_NUMBER_OF_AGGREGATION_BITS:
         case AttestationErrorCode.ERR_KNOWN_BAD_BLOCK:
         case AttestationErrorCode.ERR_AGGREGATOR_NOT_IN_COMMITTEE:
         case AttestationErrorCode.ERR_INVALID_SELECTION_PROOF:
         case AttestationErrorCode.ERR_INVALID_SIGNATURE:
-        case AttestationErrorCode.ERR_INVALID_INDEXED_ATTESTATION:
         case AttestationErrorCode.ERR_INVALID_AGGREGATOR:
-          this.logger.warn("Rejecting gossip aggregate & Proof: ", e.type);
+          this.logger.warn("Rejecting gossip aggregate and Proof: ", toJson({...e.getMetadata(), ...logContext}));
           return ExtendedValidatorResult.reject;
         case AttestationErrorCode.ERR_INVALID_SLOT_TIME:
         case AttestationErrorCode.ERR_AGGREGATE_ALREADY_KNOWN:
         case AttestationErrorCode.ERR_MISSING_ATTESTATION_PRESTATE:
-          this.logger.warn("Ignoring gossip aggregate & Proof: ", e.type);
-          return ExtendedValidatorResult.ignore;
         default:
-          break;
+          this.logger.warn("Ignoring gossip aggregate and Proof: ", toJson({...e.getMetadata(), ...logContext}));
+          return ExtendedValidatorResult.ignore;
       }
     }
-    await this.db.seenAttestationCache.addAggregateAndProof(signedAggregateAndProof.message);
-    return ExtendedValidatorResult.accept;
   };
 
   public isValidIncomingVoluntaryExit = async (
     voluntaryExit: SignedVoluntaryExit
   ): Promise<ExtendedValidatorResult> => {
     try {
       await validateGossipVoluntaryExit(this.config, this.chain, this.db, voluntaryExit);
+      return ExtendedValidatorResult.accept;
     } catch (e) {
       this.logger.error("Error while validating gossip voluntary exit");
+      if (!(e instanceof VoluntaryExitError)) {
+        this.logger.error("Gossip voluntary exit validation threw a non-VoluntaryExitError", e);
+        return ExtendedValidatorResult.ignore;
+      }
       switch (e.type.code) {
-        case ProposerSlashingErrorCode.ERR_SLASHING_ALREADY_EXISTS:
-          this.logger.warn("Ignoring gossip voluntary exit: ", e.type);
-          return ExtendedValidatorResult.ignore;
-        case ProposerSlashingErrorCode.ERR_INVALID_SLASHING:
-          this.logger.warn("Rejecting gossip voluntary exit: ", e.type);
+        case VoluntaryExitErrorCode.ERR_INVALID_EXIT:
+          this.logger.warn("Rejecting gossip voluntary exit: ", toJson(e.getMetadata()));
```

Comment:
The logger should take care of calling toJson to the data structures passed to it. @wemeetagain do you think this should be 
```
this.logger.warn("Rejecting gossip voluntary exit: ", e.getMetadata())
```
or 
```
this.logger.warn("Rejecting gossip voluntary exit: ", e)
```

Reply:
Yeah, generally speaking, I think the whole error should be logged. (And in all cases, you're right, the logger calls toJson, so it isn't needed here)

I guess the difference between the two is whether we want the stack printed or not.

Reply:
I wouldn't print stack because if node gets plenty of faulty gossip messages logs are gonna be filled with same stack.

Reply:
just pushed a commit to remove the toJson wrap.  I agree with @mpetrunic  that we don't need the stack trace for known gossip errors, but i have a stack trace for unexpected errors for each gossip topic validation.